### PR TITLE
feat(PL-2420): add view git log to release promote options

### PR DESCRIPTION
**Pull request by davidmdm**

### CODE REVIEW

File: internal/release/promote/promotion.go [hunk 1]
```diff
@@ -195,18 +197,46 @@ func (p *Promotion) Promote(opts Opts) (string, error) {
...
+		case CreateDraft:
+			performParams.draft = true
+			break loop
+		case ViewGitLog:
+			for _, item := range selectedList.Items {
+				p.println(color.InBold(fmt.Sprintf("--- %s", item.Name)))
+				source, target := at(item.Releases, sourceEnvIndex), at(item.Releases, targetEnvIndex)
+
+				info, err := getReleaseInfo(source, target, performParams)
+				if err != nil {
```
Comment (ayahajar) [inline internal/release/promote/promotion.go:220]: do you want to add a color for error?

File: internal/release/promote/promotion.go [hunk 2]
```diff
@@ -195,18 +197,46 @@ func (p *Promotion) Promote(opts Opts) (string, error) {
...
+					p.printf("error getting release information %s: %v\n", item.Name, err)
+					continue
+				}
+
+				for _, commit := range info.Commits {
+					p.printf("%s %s %s\n", color.InYellow(commit.ShortSha), color.InCyan(commit.GitHubAuthor), commit.ShortMessage)
+				}
+
+				if len(info.Commits) == 0 {
+					p.println("no commits found in service")
```
Comment (silphid) [inline internal/release/promote/promotion.go:230]:
```suggestion
					p.println("no commits found in repo")
```

File: internal/release/promote/promotion.go [hunk 3]
```diff
@@ -195,18 +197,46 @@ func (p *Promotion) Promote(opts Opts) (string, error) {
...
-		return "", nil
+		switch action {
+		case CreatePR:
+			break loop
+		case CreateDraft:
+			performParams.draft = true
+			break loop
+		case ViewGitLog:
+			for _, item := range selectedList.Items {
+				p.println(color.InBold(fmt.Sprintf("--- %s", item.Name)))
```
Comment (silphid) [inline internal/release/promote/promotion.go:216]: `style`?? 🙈

File: internal/release/promote/promotion.go [hunk 4]
```diff
@@ -195,18 +197,46 @@ func (p *Promotion) Promote(opts Opts) (string, error) {
...
+				source, target := at(item.Releases, sourceEnvIndex), at(item.Releases, targetEnvIndex)
+
+				info, err := getReleaseInfo(source, target, performParams)
+				if err != nil {
+					p.printf("error getting release information %s: %v\n", item.Name, err)
+					continue
+				}
+
+				for _, commit := range info.Commits {
+					p.printf("%s %s %s\n", color.InYellow(commit.ShortSha), color.InCyan(commit.GitHubAuthor), commit.ShortMessage)
```
Comment (silphid) [inline internal/release/promote/promotion.go:226]: `style`?? 🙉

**Review by silphid — APPROVED** [review]
LGTM, just minor comments...

File: internal/release/promote/promotion.go (hunk 4 above)
Comment (davidmdm) [reply internal/release/promote/promotion.go:226]: Hot take. We abuse style. I have no idea which styles these should be, nor if I need to create those styles, what styles I should name them. There comes a point when "style" falls apart.

File: internal/release/promote/promotion.go (hunk 3 above)
Comment (davidmdm) [reply internal/release/promote/promotion.go:216]: 🙈

**Pull request merged**